run_md.py

```diff
@@ -0,0 +1,879 @@
+
+    tu = 'ps' if mdtime_ns <= 10 else 'ns'
+    dtstep = 50 if mdtime_ns <= 10 else 100
+
+    tpr = os.path.join(wdir, f'{deffnm}.tpr')
+    xtc = os.path.join(wdir, f'{deffnm}.xtc')
+
+    try:
+        subprocess.check_output(f'wdir={wdir} index_group={index_group} tu={tu} dtstep={dtstep} deffnm={deffnm} tpr={tpr} xtc={xtc} bash {os.path.join(project_dir, "md_analysis.sh")}',
+                                shell=True)
+    except subprocess.CalledProcessError as e:
+        logging.error(f'{wdir}\t{e}\n')
+        return None
+
+    # molid resid pairs
+    # calc rmsd
+    for cur_ligand in ligands_data:
+        cur_ligand_molid, cur_ligand_resid = cur_ligand.split('\t')
+        cur_ligand_molid, cur_ligand_resid = cur_ligand_molid.strip(), cur_ligand_resid.strip()
+        md_lig_rmsd_analysis(molid=cur_ligand_molid, resid=cur_ligand_resid, wdir=wdir, tu=tu)
+
+    return wdir
+
+# def get_prev_last_time(xtc):
+#     res = subprocess.run(f'gmx check -f {xtc}', shell=True, capture_output=True)
+#     frames_step = re.findall('Step[ ]*([0-9 ]*)', res.stderr)
+#     if frames_step:
+#         frames, step = [int(i) for i in frames_step[0].split(' ') if i]
+#         frames = frames - 1
+#         time_ps = frames*step
+#         return time_ps
+
+def continue_md_from_dir(wdir_to_continue, tpr, cpt, xtc, deffnm_prev, deffnm_next, mdtime_ns, project_dir):
+    if tpr is None:
+        tpr = os.path.join(wdir_to_continue, f'{deffnm_prev}.tpr')
+    if cpt is None:
+        cpt = os.path.join(wdir_to_continue, f'{deffnm_prev}.cpt')
+    if xtc is None:
+        xtc = os.path.join(wdir_to_continue, f'{deffnm_prev}.xtc')
+
+    for i in [tpr, cpt, xtc]:
+        if not os.path.isfile(i):
+            logging.error(f'No {i} file was found. Cannot continue the simulation. Calculations will be interrupted ')
+            return None
+
+    new_mdtime_ps = int(mdtime_ns * 1000)
+
+    # check previous existing files with the same name
+    for f in glob(os.path.join(wdir, f'{deffnm_next}*')):
+        n=1
+        for _ in glob(os.path.join(wdir, f'#{os.path.basename(f)}.*#')):
+            n+=1
+        new_f = os.path.join(wdir, f'#{os.path.basename(f)}.{n}#')
+        shutil.move(f, new_f)
+        logging.warning(f'Backup previous file {f} to {new_f}')
+
+    return continue_md(tpr=tpr, cpt=cpt, xtc=xtc, wdir=wdir_to_continue,
+                       new_mdtime_ps=new_mdtime_ps, deffnm_next=deffnm_next, project_dir=project_dir)
+
+
+def continue_md(tpr, cpt, xtc, wdir, new_mdtime_ps, deffnm_next, project_dir):
+    try:
+        subprocess.check_output(f'wdir={wdir} tpr={tpr} cpt={cpt} xtc={xtc} new_mdtime_ps={new_mdtime_ps} '
+                                f'deffnm_next={deffnm_next} bash {os.path.join(project_dir, "continue_md.sh")}', shell=True)
+    except subprocess.CalledProcessError as e:
+        logging.error(f'{wdir}\t{e}\n')
+        return None
+
+    return wdir
+
+
+
+def main(protein, wdir, lfile=None, system_lfile=None,
```

Comment:
Do we need these default values for args if we supply all args explicitly? Default values of some args differ from default values of argparser.